s2spy/traintest.py

```diff
@@ -2,52 +2,94 @@
 
 A collection of train/test splitting approaches for cross-validation.
 """
+from typing import Union
+import numpy as np
 import pandas as pd
-from sklearn.model_selection import KFold
+import xarray as xr
 
 
-def kfold(df: pd.DataFrame, n_splits: int = 2, shuffle=False, **kwargs):
-    """
-    K-Folds cross-validator, which splits provided intervals into k
-    consecutive folds.
-
-    For more information
+def fold_by_anchor(cv, data: Union[xr.Dataset, pd.DataFrame]):
```

Comment:
I'm not sure that the name "fold_by_anchor" is intuitive. The anchor is quite specific to our use case with the calendar, whereas this wrapper is fairly generic. What about something like

```suggestion
def fold_grouped_data(cv, data: Union[xr.Dataset, pd.DataFrame], key: Optional[Str]="group"):
```

where `group` would be replaced by "anchor_year" for our use case?

Reply:
Additionally, `cv` is a bit of a confusing name. It's an abbreviation of "crossvalidator" or something akin, but in the docstrings we consistently refer to it as splitter class. Similarly, splitter and fold are used interchangeably. Shall we use split for everything, i.e. `split_by_anchor(splitter, data: ...)`?

Reply:
Furthermore, as a nice-to-have: I was wondering if it is possible to also give a type hint for the splitter (cv). It seems they all derive from https://github.com/scikit-learn/scikit-learn/blob/baf0ea25d6dd034403370fea552b21a6776bef18/sklearn/model_selection/_split.py#L54, can we use that?

Reply:
Finally, can we explain how this differs from skearn's existing GroupKFold/GroupShuffleSplit, and why that is useful?

Reply:
I'll clean this class up and implement your suggestions. However, as in our data the anchor_year would always be the group for our use case, let's make this the default?